ocaml/xapi/xapi_observer_components.ml

```diff
@@ -0,0 +1,106 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module Component = struct
```

Comment:
Not a big issue but do we need this module any more as we are moving to a new ml file? I think it is ok to make all of these in the `Component` module toplevel. Or does it require lots of refactoring?

Reply:
Seems like most of the changes required were in `xapi_observer.ml`. I created a fixup for this https://github.com/xapi-project/xen-api/pull/5345/commits/b9e3d635d1dfb9a2da7454245e974573fb870267.